src/test/user.test.js

```diff
@@ -0,0 +1,112 @@
+const server = require('./server')
+
+const userName = `u${Date.now()}`
+const password = `p${Date.now()}`
+const testUser = {
+    name: userName,
+    password,
+    address: "1234 Main St",
+    description: "test description"
+}
+
+let COOKIE = ''
+

```

Comment:
all the tests here only assert the `errCode` field, what benefit can we get from these tests?
there are many other styles of writing tests, e.g. test the function from the user's perspective / business perspective. 
can you compare the benefit & cost of different kinds of tests?

Reply:
ideally would be testing against possible failure cases. the implementation above obviously is too naive just a showcase of test happening. 
testing against route level I/O would be simplest but harder to narrow down the actual cause of the failure, yet providing room for refactor in a different style fashion.
testing against actual execution logic/flow would be a rather detailed approach which makes locating problem easier but when refactor happens test cases would need to be updated as well, accordingly.